docs/design/core/waiters/DecisionLog.md

```diff
@@ -0,0 +1,41 @@
+# Decision Log for SDK V2 Waiters
+
+Note: The decision log process was implemented late in this project, so decisions earlier than 9/22/20 are not included 
+below.
+
+## Log Entry Template
+
+**Source:** (Meeting/aside/pair programming discussion/daily standup) to (discuss/implement) X
+
+**Attendees:** Anna-Karin, Ben, Dongie, Irene, Matt, Nico, Vinod, John, Zoe
+
+**Closed Decisions:**
+
+1. Question? Decision. Justification.
+
+**Open Decisions:**
+
+1. (Old/Reopened/New) Question?
+
+## 9/22/20
+
+**Source:** Meeting to review API surface-area of the waiters implementation https://github.com/aws/aws-sdk-java-v2/tree/waiters-development
+
+**Attendees:** Anna-Karin, Ben, Dongie, John, Matt, Nico, Vinod, Zoe
+
+**Closed Decisions:**
+
+1. The default `ScheduledExecutorService` core number should be changed to 1 because it's only used to schedule attempts.
+2. `DynamodbEnhancedClient.builder().build()` should create a default SDK client because `DynamodbEnhancedClient.builder().build` should be equivalent to `DynamodbEnhancedClient.create()`
+3. Should we make the generic `Waiter` non-public? No, because it is essentially a protected API and has to be backwards-compatible any way. In addition, customers can benefit from it
+when service waiters are not available.
+4. We should keep the union type `ResponseOrException` because it clearly indicates response and exception are mutually exclusive and only one is present. We should keep the naming `ResponseOrException` because it's intuitive and descriptive. 
+5. We should rename `WaiterResponse#responseOrExecption` -> `WaiterResponse#matched` because it has the most votes compared with other options: `matchedResponse`, `matchedResult`, `matchedValue`
+6. We should rename `{Service}Waiter.Builder#executorService` -> `{Service}Waiter.Builder#scheduledExecutorService` to be more clear it's a `ScheduledExecutorService`
+7. We should rename `Waiter.Builder#pollingStrategy` -> `Waiter.Builder#waiterOverrideConfiguration` so that we can easily add more configurations if needed. All configs under `WaiterOverrideConfiguration` should have default values to be consistent
+with `ClientOverrideConfiguration`
+8. We should create an overload method which takes a `WaiterOverrideConfiguration` parameter to allow request-level config override for every waiter operation, eg: `DynamodbWaiter#waitUntilTableExists(DescribeTableRequest, WaiterOverrideConfiguration)`
```

Comment:
This should usually follow the "Question? Decision. Justification." format, even if we decide not to change anything.

Reply:
Might be good to have an official scribe for the future meetings, so they can capture questions, even if they don't have action items. It seems like these are just the questions we decided to act on.

Reply:
+1

Reply:
I'll update it to question format